[glean] Document the pings available out of the box [ci skip] (#1566)
mozilla-mobile/android-components: components/service/glean/docs/baseline.md

# The `baseline` ping
This ping is intended to provide metrics that are managed by the library itself, and not explicitly

> set from the application or included in the application's `metrics.yaml` file.

Reviewer: ```suggestion
set by the application or included in the application's `metrics.yaml` file.
```

Author: Thanks Chris, this is very helpful! I'm making the changes locally and then I'll force-push here (vs accepting the changes here online).

> The `baseline` ping is automatically sent when the application is moved to *background* and it includes

Reviewer: ```suggestion
The `baseline` ping is automatically sent when the application is moved to the *background* and it includes
```

the following fields:

| Field name | Type | Description |
|---|---|---|
| `sessions` | Counter | The number of foreground sessions since the last upload |
| `durations` | Timespan | The combined duration of all the foreground sessions since the last upload |
| `os` | String | The name of the operating system (e.g. "linux", "android", "ios") |
| `os_version` | String | The version of the operating system |
| `device` | String | Manufacturer and model |
| `architecture` | String | The architecture of the device (e.g. "arm", "x86") |

> | `timezone` | Number | The timezone of the device, in offset from UTC |

Reviewer: ```suggestion
| `timezone` | Number | The timezone of the device, as an offset from UTC |
```

> | `accessibility_services` | StringList | List of accessibility services enabled on the device |

Reviewer: This is on the `baseline` ping? ...that's interesting. I wouldn't have expected this to be baseline data.

Author: Yeah :E

> | `locale` | String | The locale of the application |

Reviewer 2: I'm a bit worried that this table and the descriptions in the `metrics.yaml` will get out of sync, and sort of feel that the `metrics.yaml` should be the one source of truth.  But that's probably going to require a kotlindoc plugin that I'd rather not write, so we should probably not worry about it for now.

Reviewer: It's tricky because we have a process requirement here to be clear about the things glean collects and why. If we could embed the relevant portions it would get the job done but it would also mean that you don't have to consider keeping the docs up to date when you add new baseline metrics.

It may be that we find that "it's hard to keep the two up-to-date" is a feature not a bug, forcing us to be more deliberative and process-heavy when we add collections to this core library.

Author: @mdboom I see your concern. On the other hand, the fact that new collection *must* require a data-review could be a reminder for keeping things in sync :)
